services/QuillLMS/app/controllers/sales_form_submission_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SalesFormSubmissionController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  RENEWAL_REQUEST = 'renewal request'
+  QUOTE_REQUEST = 'quote request'
+  SCHOOL = 'school'
+  DISTRICT = 'district'
+
+  def request_renewal
+    @type = RENEWAL_REQUEST
+  end
+
+  def request_quote
+    @type = QUOTE_REQUEST
+  end
+
+  def create
+    sales_form_submission = SalesFormSubmission.new(sales_form_submission_params)
+    if  sales_form_submission.save!
+      head :no_content, status: 200
+    else
+      render json: sales_form_submission.errors, status: :unprocessable_entity
+    end
+  end
+
+  def options_for_sales_form
+    type = params[:type]
+    schools_or_districts = type == SCHOOL ? School.all : District.all
+    options = schools_or_districts.map do |school_or_district|
+      {
+        name: school_or_district.name,
+        value: school_or_district.name
+      }
+    end
```

Comment:
This is a matter of style mostly, but I'd be inclined to say that the backend should just pass up the data and then the frontend should format it however it needs to pass it to the rendering component (so the mapping should happen there, not here). The backend should be agnostic to how the data gets used.

Reply:
ok, that's a fair point. I can just pass up an array of all the schools and districts

Reply:
If you're just using names (and don't care about the objects themselves), you could use:
`options = type.classify.constantize.all.pluck(:name)`